### PR TITLE
feat: use system page properties

### PR DESCRIPTION
**Pull request by kchung**
## Changes
* Transition System Pages to use Page properties

This required defining the pagination and sorting logic to the templates

### CODE REVIEW

File: site/pages/category.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "route": "shop/:name/:id",
+    "template": "templates/store/category",
+    "props": {
+        "category": {
+            "filters": {
+                "id": "request_data(path.id)"
```
Comment (ayee-square) [inline site/pages/category.json]: We'll be requiring `square_online_id:true` soon
```suggestion
                "id": "request_data(path.id)",
                "square_online_id": true
```

File: theme/components/sections/store/item-list.html.twig [hunk 2]
```diff
@@ -9,6 +9,45 @@
...
+{# Pluck item filters from query #}
+{% set queryFilters = ['price_min', 'price_max', 'location_id', 'has_discounts'] %}
+{% for filter in queryFilters %}
+    {% if request.query[filter] %}
+        {% set items_options = items_options|merge({ (filter): request.query[filter] }) %}
+    {% endif %}
+{% endfor %}
+
+{# Pluck comma separated item filters from query and convert to array #}
+{% set commaQueryFilters = ['fulfillments', 'category_ids', 'item_status', 'option_choices'] %}
```
Comment (ayee-square) [inline theme/components/sections/store/item-list.html.twig:22]: We'll need `square_online_id:true` when the `category_ids` filter is applied

File: theme/templates/store/category.html.twig [hunk 3]
```diff
@@ -27,6 +27,9 @@
         }) }}
     {% endif %}
 
+    {# Items options #}
+    {% set itemsOptions = { category_id: category.square_online_id } %}
```
Comment (ayee-square) [inline theme/templates/store/category.html.twig]:
```suggestion
    {% set itemsOptions = { category_id: category.square_online_id, square_online_id: true } %}
```

File: theme/templates/store/category.html.twig [hunk 4]
```diff
@@ -43,10 +46,10 @@
 
     {# Category items #}
     {{ include('components/sections/store/item-list', {
-        items,
         category,
         categories,
         category_hierarchy: category_hierarchy(null, category.square_online_id),
```
Comment (ayee-square) [inline theme/templates/store/category.html.twig]:
```suggestion
        category_hierarchy: category_hierarchy(null, category.square_online_id, true),
```

third param is square_online_id

**Review by bashunaimiroy — APPROVED** [review]

**Pull request merged**